tests/integration/test_partition_evolution.py

```diff
@@ -564,3 +564,28 @@ def _validate_new_partition_fields(
     assert len(spec.fields) == len(expected_partition_fields)
     for i in range(len(spec.fields)):
         assert spec.fields[i] == expected_partition_fields[i]
+
+
+@pytest.mark.integration
```

Comment:
thanks for this test! could you add another one to check that the same check is applied with creating the table with both schema and partition spec? 

something like
```
    schema = Schema(
        NestedField(1, "id", LongType(), required=False),
        NestedField(2, "event_ts", TimestampType(), required=False),
        NestedField(3, "another_ts", TimestampType(), required=False),
        NestedField(4, "str", StringType(), required=False),
    )
   partition_spec = PartitionSpec(
        PartitionField(source_id=1, field_id=1000, transform=IdentityTransform(), name="id"), spec_id=1
    )
    table = _create_table_with_schema(catalog, schema, "2", partition_spec)
```